enterprise/cmd/frontend/internal/batches/resolvers/batch_spec_workspace_file.go

```diff
@@ -101,7 +102,7 @@ func (r *batchSpecWorkspaceFileResolver) RichHTML(ctx context.Context) (string,
 }
 
 func (r *batchSpecWorkspaceFileResolver) URL(ctx context.Context) (string, error) {
-	return "", errors.New("not implemented")
+	return fmt.Sprintf("/files/batch-changes/%s/%s", r.file.RandID, r.batchSpecRandID), nil
```

Comment:
```suggestion
	return fmt.Sprintf("/files/batch-changes/%s", r.file.RandID), nil
```

> {bucket}/{key}

Since we are going with the `{bucket}/{key}` signature - we can safely remove the `batchSpecRandID` passed here.

Reply:
Does the workspace resolver support that?

Reply:
@eseliger We don't use that ID - yes, there's a foreign key constraint on the `batch_spec_workspace_file` table but we don't consciously use that in the handlers.

It's probably something that'd be nice to have, now that I think about it.

Please ignore my suggestion @adeola-ak